albumentations/core/composition.py

```diff
@@ -269,6 +269,61 @@ def __call__(self, force_apply=False, **data):
         return data
 
 
+class SomeOf(BaseCompose):
+    """Select some of the transforms to apply. Selected transforms will be called with `force_apply=True`.
+    Transforms probabilities will be normalized to one 1, so in this case transforms probabilities works as weights.
+
+    Args:
+        sample_range (int, int): Range of possible sample sizes.
+        transforms (list): list of transformations to compose.
+        p (float): probability of applying selected transform. Default: 0.5.
+    """
+
+    def __init__(self, sample_range: (int, int), transforms, p=0.5):
+        super().__init__(transforms, p)
+        if isinstance(sample_range, (list, tuple)):
+            self.sample_range = sample_range
+        elif isinstance(sample_range, int):
+            self.sample_range = (0, sample_range)
+        else:
+            raise TypeError(f'Invalid type for sample_range: {type(sample_range).__name__}')
+        self.sample_range = sample_range
+        assert len(self.sample_range) == 2
+        assert self.sample_range[0] <= self.sample_range[1]
+        assert all([isinstance(val, int) for val in self.sample_range])
+        assert self.sample_range[1] <= len(transforms)
+    
+    def _to_dict(self) -> dict:
+        item_dict = super()._to_dict()
+        item_dict['sample_range'] = self.sample_range
+        return item_dict
+
+    def __call__(self, force_apply=False, **data):
+        if self.replay_mode:
```

Comment:
As a rule of thumb, generation of all parameters (sampling indexes) should be done in `get_params`. This makes the __call__ function deterministic (you simply get the indexes of transforms to apply) and compatible with A.ReplayCompose. Current implementation will not work in Replay mode.

Reply:
I'm sorry, I'm not quite familiar with how replay mode works or what its purpose is.
I looked at the ReplayCompose code as well as some of the documentation, and I gather that it is used to record information about which augmentations were applied and which parameters were used.
If I understand what you're saying correctly, I would need to make it so that the indexes of the applied transforms are recorded during replay mode. I am a bit confused, though, since the A.OneOf class doesn't seem to do this either.
I copied this if statement from the A.OneOf class.
Would creating a get_params method under A.SomeOf that returns {"used_idx_list": used_idx_list} and using that in __call__ satisfy your request?

Reply:
@BloodAxe Library uses another logic for `ReplayCompose`. `ReplayCompose` checks is `params` is `None` for each transformation. If `params` is `None` - this transformation is not called.
https://github.com/albumentations-team/albumentations/blob/ba44effb0369ba5eae1e8eb4909105eac9709230/albumentations/core/composition.py#L389